debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/tests/mariadb/MariaDBTests.java

```diff
@@ -0,0 +1,178 @@
+import io.debezium.testing.system.tools.databases.mysql.MySqlController;
+import io.debezium.testing.system.tools.kafka.ConnectorConfigBuilder;
+import io.debezium.testing.system.tools.kafka.KafkaConnectController;
+import io.debezium.testing.system.tools.kafka.KafkaController;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+
+import static io.debezium.testing.system.assertions.KafkaAssertions.awaitAssert;
+import static io.debezium.testing.system.tools.ConfigProperties.DATABASE_MARIADB_PASSWORD;
+import static io.debezium.testing.system.tools.ConfigProperties.DATABASE_MARIADB_USERNAME;
+import static io.debezium.testing.system.tools.ConfigProperties.DATABASE_MYSQL_PASSWORD;
+import static io.debezium.testing.system.tools.ConfigProperties.DATABASE_MYSQL_USERNAME;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class MariaDBTests extends ConnectorTest {
+
+    public MariaDBTests(
+                      KafkaController kafkaController,
+                      KafkaConnectController connectController,
+                      ConnectorConfigBuilder connectorConfig,
+                      KafkaAssertions<?, ?> assertions) {
+        super(kafkaController, connectController, connectorConfig, assertions);
+    }
+
+    public void insertCustomer(
+                               SqlDatabaseController dbController,
+                               String firstName, String lastName,
+                               String email)
+            throws SQLException {
+        SqlDatabaseClient client = dbController.getDatabaseClient(DATABASE_MARIADB_USERNAME, DATABASE_MARIADB_PASSWORD);
+        String sql = "INSERT INTO customers VALUES  (default, '" + firstName + "', '" + lastName + "', '" + email + "')";
+        client.execute("inventory", sql);
+    }
+
+    public void renameCustomer(SqlDatabaseController dbController, String oldName, String newName) throws SQLException {
+        SqlDatabaseClient client = dbController.getDatabaseClient(DATABASE_MARIADB_USERNAME, DATABASE_MARIADB_PASSWORD);
+        String sql = "UPDATE customers SET first_name = '" + newName + "' WHERE first_name = '" + oldName + "'";
+        client.execute("inventory", sql);
+    }
+
+    public int getCustomerCount(SqlDatabaseController dbController) throws SQLException {
+        SqlDatabaseClient client = dbController.getDatabaseClient(DATABASE_MARIADB_USERNAME, DATABASE_MARIADB_PASSWORD);
+        String sql = "SELECT count(*) FROM customers";
+        return client.executeQuery("inventory", sql, rs -> {
+            try {
+                rs.next();
+                return rs.getInt(1);
+            }
+            catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    @Order(10)
+    public void shouldHaveRegisteredConnector() {
```

Comment:
We should possibly have a base class `BinlogDBTests` which would contain these tests and this class and `MySqlTests` inheriting from it to avoid code duplication.

Reply:
Sure, this is just an initial implementation.

Reply:
The idea of abstracting those tests to something like the BinlogDBTests class sounds great, but it will require major changes in the rest of the test suite's functionalities. You will need to have one database controller, one credential, and so on. I think we should merge this PR and maybe create an issue to look into in the future.

Reply:
I see. I'll eventually take a look on it and created DBZ-8309 not to forget on it.